### ISSUE TITLE
Failed connection causes panic

### ISSUE THREAD

**Issue by qwerty2586** [monaco-io/request#23]
Hello, you have great library there. I found something very unsettling. If connection doesnt happen(host down), calling sugar.Code() causes nil pointer panic because Response member is nil. Would you consider preventing that? Something like 
```
func (s *Sugar) Code() int {
  if s.ctx.Response == nil {
    return 0
  }
  return s.ctx.Response.StatusCode
}
```

**Comment by luxuze** [comment]
ok，you can add this nil pointer check and take a pull request

**Comment by luxuze** [comment]
pr merged, v1.0.15 released ~
